### PR TITLE
Improve mobilenetv2 INT8 performance by using INT8 relu as post-op

### PR DESCRIPTION
**Pull request by lidanqing-vv**
This PR improved mobilenetv2 INT8 performance with good accuracy.

We have to use relu instead of brelu as the post-op in INT8 conv2d kernel , because INT8 brelu as a post-op is not enabled in mkldnn v01.8. I add TODO and comments of what will be changed when v0.20 is enabled.

The performance of mobilenetv2 with this PR is follows:

INT8/FP32 | Top1 Accuracy |  Performance
| -- |-- | --
FP32 | 71.90% | X
INT8 | 71.43% | 1.92 X

test machine: Intel(R) Core(TM) i9-7940X CPU @ 3.10GHz, 14 Cores
paddle_num_threads 14

Performance of all the int8 models on CLX and SKX will be delivered by this Friday.

### CODE REVIEW

**Comment by lidanqing-vv** [comment]
Because I wanted to remove the conflicted commits in [PR17546](https://github.com/PaddlePaddle/Paddle/pull/17546) and I did something like renaming. Somehow previous PR was closed automatically.
I am refering to previous PR's reviews and commit new changes according to that [PR17546](https://github.com/PaddlePaddle/Paddle/pull/17546) reviews.

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (luotao1) [inline paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc]: How about using a variable to compute `fuse_relu || fuse_brelu` at first?
`bool xxx = fuse_relu || fuse_brelu`

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (luotao1) [inline paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc]: `false, 0.0`, please see 'It will be removed once the int8 is enabled.' https://github.com/PaddlePaddle/Paddle/pull/17130#discussion_r285882103

Comment (lidanqing-vv) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc]: > `false, 0.0`, please see 'It will be removed once the int8 is enabled.' [#17130 (comment)](https://github.com/PaddlePaddle/Paddle/pull/17130#discussion_r285882103)

Hi, @luotao1 I changed but do not merge and do not do further review. Previous PR worked, but after merging some updates, there seems some error. Let me test locally first.

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (lidanqing-vv) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc]: > How about using a variable to compute `fuse_relu || fuse_brelu` at first?
> `bool xxx = fuse_relu || fuse_brelu`

Yes. I agree with you. I will change. It is about using uint8_t or int8_t

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (lidanqing-vv) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc]: I  just found CreatePostOps made problems. As in https://github.com/lidanqing-intel/Paddle/commits/develop-mobilnetv2, only last commit makes the accuracy 0. All previous commits are giving good accuracy. I will fix during the day, also will talk with @wojtuss.

**Review by wojtuss — APPROVED** [review]

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (luotao1) [inline paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:470]: 470: `fuse_relu || fuse_brelu`-》`unsigned_output` and why notes `/*fuse_relu*/`

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (luotao1) [inline paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:477]: 477: `fuse_relu || fuse_brelu`-》`unsigned_output` and why notes `/*fuse_relu*/`

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (luotao1) [inline paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:478]: do you need `false /*fuse_brelu*/` again since you have `fuse_relu || fuse_brelu` before?

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (luotao1) [inline paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:471]: do you need `false /*fuse_brelu*/` again since you have `fuse_relu || fuse_brelu` before?

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (luotao1) [inline paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:479]: Could you combine `ConvFwdPrimitiveDesc` two functions, `if (bias) = false`, pass `bias_md=nullptr` into it.

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (lidanqing-vv) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:470]: @luotao1  Hi, as in comments [line 460](https://github.com/PaddlePaddle/Paddle/pull/17570/files#diff-ff3a710419e42895ac7587353a7af5bcR460). When mkldnn v0.20 is enabled, the INT8 brelu post-ops will be supported, I will substitute with the code noted in `/**/`. 

 I think we should not substitute `fuse_relu || fuse_brelu` with `unsigned_output`, in fact, we should use `fuse_relu` only at this position. We use `fuse_relu || fuse_brelu` now because in INT8 inference and achieve good accuracy, just because `relu` and `brelu` both give unsigned in inference. But the correct way is the code in `/**/`.

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (lidanqing-vv) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:479]: > Could you combine `ConvFwdPrimitiveDesc` two functions, `if (bias) = false`, pass `bias_md=nullptr` into it.

Ok, I will do like this.

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (luotao1) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:470]: Got it, when we can update to mkldnn v0.20?

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (luotao1) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:479]: I merge your pr at first, and you can refine in next PR!

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (lidanqing-vv) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:470]: > Got it, when we can update to mkldnn v0.20?

mkldnn v0.20 code freeze is 7th of June, but they need time for testing and release date is by 28th of June.

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc
Comment (lidanqing-vv) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:479]: > I merge your pr at first, and you can refine in next PR!

Thank you! I start to refine now and will check the refined code through tests. I just update that it could be mkldnn v1.0 that will be used by baidu. Both support this op and both will be released after paddle release 1.5.

Comment (luotao1) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:479]: Got it. But the refined code PR could be created before paddle release 1.5?

Comment (lidanqing-vv) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc:479]: > Got it. But the refined code PR could be created before paddle release 1.5?

Yes, will be before 1.5

**Pull request merged**
